src/Evolution/Initialization/Evolution.hpp

```diff
@@ -159,7 +159,10 @@ struct Evolution {
 
     return std::make_tuple(
         merge_into_databox<Evolution, simple_tags, compute_tags>(
-            std::move(box), TimeId{}, time_id, initial_dt, std::move(dt_vars),
+            std::move(box),
+            // Use the initial time id also as the next. When initializing the
+            // time stepper history the next time id will be set properly.
+            time_id, time_id, initial_dt, std::move(dt_vars),
```

Comment:
Note to reviewers: This change makes sure the `temporal_id` in the DataBox is actually the initial time, so that `InitializeMortars` can read it (instead of the `::Tags::Next<temporal_id>` that was read before).

Reply:
Why?  What was wrong with using `::Tags::Next<temporal_id>`?  Particularly since the thing you're initializing is `::Tags::Mortars<::Tags::Next<temporal_id>, dim>`, so it strikes me as more natural to use the `Next` version.

Reply:
I didn't touch what is being initialized, since the fluxcomm code depends on it. It uses `::Tags::Mortars<::Tags::Next<temporal_id>, dim>`, so that's what I'm initializing. Perhaps when refactoring the fluxcomm code that can be reconsidered, but shouldn't be part of this PR. Now to the initialization procedure:
- Situation before this PR: `temporal_id` was initialized to an invalid value and `::Next<temporal_id>` was initialized to the actual initial time. Then the first call to `AdvanceTime` (which occurs in the next phase) takes care of moving the initial time to `temporal_id` and sets the correct `::Next<temporal_id>`. Since the mortars are initialized before `AdvanceTime` is called, they pull the initial time value out of `::Tags::Next<temporal_id>`.
- Situation with this PR: `temporal_id` is initialized to the actual initial time. Also `::Next<temporal_id>` is set to that time, because I think the self-starting procedure assumes it. To initialize the mortars, we can now pull the initial time from `temporal_id` instead of `::Next<temporal_id>`.

This not only makes more sense to me, but in the elliptic code I don't have a self-starting procedure so I just initialize `temporal_id` to the initial step number and `::Next<temporal_id>` to that incremented by one. So the mortars initializer should pull the initial temporal id from `temporal_id`, not `::Next<temporal_id>`.

Reply:
`::Tags::Mortars<::Tags::Next<temporal_id>, dim>` is the correct tag.  In evolution, it represents the next TimeId on the neighbor across the mortar.  Since initially all the elements are in sync and have the same time step, the next TimeId in each neighbor is the same as the next TimeId in the element itself, so the mortar `Next<TimeId>` should be set to the element `Next<TimeId>`.

This doesn't require self-starting in order to make sense, and the current method predates the self-staring code.  (Although it was done this way with the goal of making that transition easier.)

If this is incompatible with the elliptic code then we may need separate initialization actions for the two modes.  You've already had to add a hack to make the elliptic code compatible with  the LTS mortar code.

Reply:
I'm in agreement with @wthrowe here.

Reply:
You don’t initialize the mortars to the _next_ time ID though, but you initialize them to the _initial_ time ID, which happens to be in the `::Tags::Next<temporal_id>` at that point.

Reply:
To be clear: When the time starts at `0.0`, the time that should be set on the mortars is also `0.0`. Same for elliptic code: the iteration number starts at `0` and also the mortars should be initialized at `0`.